### PR TITLE
Untangle: Move Settings > Newsletter to Jetpack

### PR DESCRIPTION
**Pull request by lupus2k**
Fixes https://github.com/Automattic/dotcom-forge/issues/5461

## Proposed changes:
Move Newsletter submenu from Settings to Jetpack, and show it under `Settings` or `Subscribers` (if https://github.com/Automattic/jetpack/pull/35429 is merged), and add an external link icon.

| Before | After |
| --- | --- |
| image | image |

## Jetpack product discussion
Details on Issue: https://github.com/Automattic/dotcom-forge/issues/5445

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Set your `Admin interface style` to `Classic (wp-admin)` in `wordpress.com/hosting-config/site_slug`
* Go to `wp-admin`
* The Jetpack menu should have the Newsletter item with an external link icon.
* The Settings menu should not have the Newsletter item.

### CODE REVIEW

**Review by DustyReagan — DISMISSED** [review]
LGTM!

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 1]
```diff
@@ -336,6 +336,10 @@ public function add_jetpack_menu() {
 		// This is supposed to be the same as class-admin-menu but with a different position specified for the Jetpack menu.
 		if ( 'wp-admin' === get_option( 'wpcom_admin_interface' ) ) {
 			parent::create_jetpack_menu( 2, false );
+
+			// Remove the Newsletter submenu item from Settings and add it under Jetpack menu.
+			remove_submenu_page( 'options-general.php', 'https://wordpress.com/settings/newsletter/' . $this->domain );
```
Comment (lsl) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Do you think we should wrap [this line](https://github.com/automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L384) with the wp-admin check instead or do you prefer keeping this code together?

Comment (fushar) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Does this change apply to Simple sites as well? If so, I think we should move this check to the base class.

Comment (arthur791004) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Looks like we have different ways to untangle the menu/submenu. Maybe we have to unify the approach 🤔
* Wrap the add_submenu_page with the wp-admin check
* Use the `remove_submenu_page` and `hide_submenu_page` somewhere

Comment (fushar) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Yes, it is related to where to put the logic:

- Applicable to both Atomic and Simple? Wrap the `add_submenu_page()` with the check in the admin-menu class
- Applicable to only Atomic? Use `remove_submenu_page()` in the atomic-admin-menu class

Comment (lupus2k) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: True, I did it thinking only on Atomic, should move it for Simple+Atomic?

Comment (lupus2k) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: I updated with your suggestions

**Comment by fushar** [comment]
It conflicts with https://github.com/Automattic/jetpack/pull/35429; let's merge that one first and then rebase this one 😄

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php (hunk 1 above)
Comment (lsl) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Same as https://github.com/Automattic/jetpack/pull/35429#discussion_r1480791682 classic view only for now.

**Review by candy02058912 — APPROVED** [review]
LGTM 👍

**Pull request merged**